plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowDependencyManager.java

```diff
@@ -125,7 +125,23 @@ public boolean hasTrackedDataflowDependency(IProject project) {
         || Artifact.RELEASE_VERSION.equals(dependency.getVersion()));
   }
 
+  /**
+   * Returns a possible version range for the given project.
+   * 
+   * @throws IllegalStateException if the version range is invalid
+   */
   public VersionRange getDataflowVersionRange(IProject project) {
+    VersionRange actual = getActualDataflowVersionRange(project);
+    return actual != null ? actual : allVersions();
+  }
+
+  /**
+   * Returns the encoded version range for the given project or {@code null} if not found.
+   * 
+   * @return the version range or {@code null} if not found
+   * @throws IllegalStateException if the version range is invalid
```

Comment:
I think the Javadoc should be something callers can understand. Unclear to me as a caller when a range is invalid and how to avoid it.